firmware/library/L4_Application/task_scheduler_test.cpp

```diff
@@ -0,0 +1,129 @@
+// Tests for the TaskScheduler singleton class.
+#include "L4_Application/task.hpp"
+#include "L5_Testing/testing_frameworks.hpp"
+
+FAKE_VOID_FUNC(vTaskStartScheduler);
+FAKE_VOID_FUNC(vTaskSuspend, TaskHandle_t);
+FAKE_VOID_FUNC(vTaskResume, TaskHandle_t);
+FAKE_VOID_FUNC(vTaskDelete, TaskHandle_t);
+FAKE_VOID_FUNC(vTaskDelayUntil, TickType_t *, TickType_t);
+FAKE_VOID_FUNC(vApplicationGetIdleTaskMemory, StaticTask_t **, StackType_t **,
+               uint32_t *);
+FAKE_VALUE_FUNC(TickType_t, xTaskGetTickCount);
+FAKE_VALUE_FUNC(TaskHandle_t, xTaskCreateStatic, TaskFunction_t, const char *,
+                uint32_t, void *, UBaseType_t, StackType_t *, StaticTask_t *);
+FAKE_VALUE_FUNC(EventGroupHandle_t, xEventGroupCreateStatic,
+                StaticEventGroup_t *);
+FAKE_VALUE_FUNC(EventBits_t, xEventGroupSync, EventGroupHandle_t, EventBits_t,
```

Comment:
Should these be place in a header file for reuse?

Reply:
Yes, they should be. The rtos.hpp file would be a great place to put these. 

You can do something like this:
```
#if HOST_TEST
// insert the task fakes
#else
#include "FreeRTOS.h"
#endif  // HOST_TEST 
```